src/components/CourseCard/BaseCourseCard.jsx

```diff
@@ -3,53 +3,55 @@ import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import moment from 'moment';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-
-import './CourseCard.scss';
+import { StatusAlert } from '@edx/paragon';
 
 import EmailSettingsModal from './EmailSettingsModal';
 
+import './CourseCard.scss';
+
 class BaseCourseCard extends Component {
   state = {
     modals: {
-      email_settings: null,
+      emailSettings: null,
     },
+    hasEmailsEnabled: this.props.hasEmailSettings || false,
+    hasNewEmailSettings: false,
   };
 
   setModalState = ({ key, options }) => {
-    this.setState({
+    this.setState(state => ({
```

Comment:
nit: If you need the existing state in `this.setState()`, it's best to pass in a function which receives the current state as a function argument.